bigtable/client/readrowsparser.h

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_READROWSPARSER_H_
+#define BIGTABLE_CLIENT_READROWSPARSER_H_
+
+#include "bigtable/client/version.h"
+
+#include "bigtable/client/cell.h"
+#include "bigtable/client/row.h"
+
+#include "google/bigtable/v2/bigtable.grpc.pb.h"
+
+#include <absl/strings/string_view.h>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * The internal module responsible for transforming ReadRowsResponse
+ * protobufs into Row objects.
+ *
+ * Users are expected to do something like:
+ *   while (!stream.EOT()) {
+ *     chunk = stream.NextChunk();
+ *     parser.HandleChunk(chunk);
+ *     if (parser.HasNext()) {
+ *       row = parser.Next();  // you now own `row`
+ *     }
+ *   }
+ *   parser.HandleEOT();
+ */
+class ReadRowsParser {
+ public:
+  /**
+   * Pass an input chunk proto to the parser. May throw errors, in
+   * which case valid data read before the error is still accessible.
+   */
+  void HandleChunk(
+      const google::bigtable::v2::ReadRowsResponse_CellChunk& chunk);
+
+  /**
+   * Signal that the input stream reached the end. May throw errors if
+   * more data was expected, in which case valid data read before the
+   * error is still accessible.
+   */
+  void HandleEOT();
+
+  /**
+   * True if the data parsed so far yielded a Row. Call Next() to take
+   * the row. Throws an error if there is wrong or incomplete data in
```

Comment:
Please use the `@throws` doxygen command to describe what exceptions may be raised.

Reply:
done